packages/hardhat-core/src/internal/core/providers/http.ts

```diff
@@ -58,6 +58,10 @@ export class HttpProvider extends EventEmitter implements EIP1193Provider {
       throw error;
     }
 
+    if (args.method === "hardhat_reset") {
+      this.emit("hardhat_reset");
```

Comment:
Not sure about the name of this event.

Reply:
maybe `hardhatNetwokReset`? I'd create a constant.